### ISSUE TITLE
Get Users in specified role(s)

### ISSUE THREAD

**Issue by chrisbonilla95** [mozts2005/ZendeskApi_v2#300]
Is there a way to get all the users in a certain role? Can't seem to find it and pulling all users and then filtering is quite inefficient when you have a lot of end-users. I for example just want to currently pull agents and admins

I found this on zendesk https://developer.zendesk.com/rest_api/docs/core/users#filters

**Comment by chrisbonilla95** [comment]
Don't know if this is the cleanest way to do it, but could you not just add a parameter to GetAllUsers that takes a list of roles (ex. List<string> roles = null) and then modify the method as follows:

```
public GroupUserResponse GetAllUsers(int? perPage = null, int? page = null, UserSideLoadOptions sideLoadOptions = UserSideLoadOptions.None, List<string> roles = null)
        {
            string resourceString = "users.json";

            if (roles != null && roles.Count > 0)
            {
                resourceString += "?";

                foreach (var role in roles)
                {
                    resourceString += "role[]=" + role + "&";
                }

                resourceString.TrimEnd('&');
            }
            
            string resource = GetResourceStringWithSideLoadOptionsParam(resourceString, sideLoadOptions);

            return GenericPagedGet<GroupUserResponse>(resource, perPage, page);
        }
```
I tried building to test, but I keep getting errors about versions strings and metadata.

**Comment by chrisbonilla95** [comment]
So I figured out building it and added the dll to my project to test and can confirm that it works.
I also tried sending just 1 role since it says to use role[]= for more than 1 and role= for just 1 and role[] works just fine for just 1 role sent as well, so that should probably be used since it'll work for either scenario.

I just added to that method because it was easy/quick, but perhaps it makes sense to create separate methods, like GetAllAgents, GetAllEndUsers, GetAllAdmins and GetAllUsersInRoles. 

Just my 2 cents, thanks for all you're hard work, makes my life much easier!
